File: app/src/main/java/com/example/hotfix/note/class11/gson/Course.java
package com.example.hotfix.note.class11.gson;

public class Course {

    private String name;
    private float score;

    public Course() {
        System.out.println("_Course: empty");
    }

    public Course(String name, float score) {
        System.out.println("_Course: " + name + " " + score);
        this.name = name;
        this.score = score;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public float getScore() {
        return score;
    }

    public void setScore(float score) {
        this.score = score;
    }

    @Override
    public String toString() {
        return "Course{" +
                "name='" + name + '\'' +
                ", score=" + score +
                '}';
    }
}

File: app/src/main/java/com/example/hotfix/note/class10/Course.kt
package com.example.hotfix.note.class10

import java.io.*

data class Course(var name: String, var score: Float) : Serializable {
    companion object {
        private const val serialVersionUID: Long = 1L
    }
}

private fun test1() {
    var course = Course("jianruilin", 100f)

    var out = ByteArrayOutputStream()

    var oos = ObjectOutputStream(out)

    println("序列化前$course")

    oos.writeObject(course)

    var bs = out.toByteArray()

    oos.close()

    var ois = ObjectInputStream(ByteArrayInputStream(bs))

    course = ois.readObject() as Course

    println("序列化后$course")
}

fun main() {
    test1()
}